lib_nbgl/src/nbgl_use_case.c

```diff
@@ -1526,5 +1547,197 @@ void nbgl_useCaseSpinner(const char *text)
+ * @param backToken token used with actionCallback (0 if unused))
+ * @param shuffled if set to true, digits are shuffled in keypad
+ * @param tuneId if not @ref NBGL_NO_TUNE, a tune will be played when back button is pressed
+ * @param validatePinCallback function calledto validate the pin code
+ * @param onActionCallback callback called on any action on the layout
+ */
+void nbgl_useCaseKeypad(const char                *title,
+                        uint8_t                    minDigits,
+                        uint8_t                    maxDigits,
+                        uint8_t                    backToken,
+                        bool                       shuffled,
+                        tune_index_e               tuneId,
+                        nbgl_pinValidCallback_t    validatePinCallback,
+                        nbgl_layoutTouchCallback_t actionCallback)
+{
+    nbgl_layoutDescription_t  layoutDescription = {0};
+    nbgl_layoutCenteredInfo_t centeredInfo      = {0};
+    int                       status            = -1;
+
+    if ((minDigits > KEYPAD_MAX_DIGITS) || (maxDigits > KEYPAD_MAX_DIGITS)) {
+        return;
+    }
+
+    reset_callbacks();
+    // reset the keypad context
+    memset(&keypadContext, 0, sizeof(KeypadContext_t));
+
+    // get a layout
+    layoutDescription.onActionCallback = actionCallback;
+    layoutDescription.modal            = true;
+    layoutDescription.withLeftBorder   = false;
+    keypadContext.layoutCtx            = nbgl_layoutGet(&layoutDescription);
+
+    // set navigation bar
+    nbgl_layoutAddProgressIndicator(
+        keypadContext.layoutCtx, 0, 0, (backToken != 0), backToken, tuneId);
+
+    // add text description
+    centeredInfo.text1 = title;
+    centeredInfo.style = LARGE_CASE_INFO;
+    centeredInfo.onTop = true;
+    nbgl_layoutAddCenteredInfo(keypadContext.layoutCtx, &centeredInfo);
+
+    // add keypad
+    status = nbgl_layoutAddKeypad(keypadContext.layoutCtx, nbgl_keypadCallback, shuffled);
+    if (status < 0) {
+        nbgl_layoutRelease(keypadContext.layoutCtx);
+        keypadContext.layoutCtx = NULL;
+        return;
+    }
+    keypadContext.keypadIndex = (unsigned int) status;
+
+    // add hidden digits
+    status = nbgl_layoutAddHiddenDigits(keypadContext.layoutCtx, maxDigits);
+    if (status < 0) {
+        nbgl_layoutRelease(keypadContext.layoutCtx);
+        keypadContext.layoutCtx = NULL;
+        return;
+    }
+    keypadContext.hiddenDigitsIndex = (unsigned int) status;
+
+    // validation pin callback
+    onValidatePin = validatePinCallback;
+    // pin code acceptable lengths
+    keypadContext.pinMinDigits = minDigits;
+    keypadContext.pinMaxDigits = maxDigits;
+
+    nbgl_layoutDraw(keypadContext.layoutCtx);
+    nbgl_refreshSpecialWithPostRefresh(FULL_COLOR_CLEAN_REFRESH, POST_REFRESH_FORCE_POWER_ON);
+}
+
+/**
+ * @brief keypad layout release
+ */
+void nbgl_useCaseKeypadRelease(void)
```

Comment:
Why a release for Keypad ? Use Case layer doesn't require it....

Reply:
Ok, removed

Reply:
Indeed, it is needed, else, the screen is kept in background (not released)

Reply:
Ok, removing `modal`, and now the release is no more needed.
Fixed!